extension/chrome/elements/compose-modules/formatters/signed-msg-mail-formatter.ts

```diff
@@ -34,12 +34,15 @@ export class SignedMsgMailFormatter extends BaseMailFormatter {
       const allContacts = [...newMsg.recipients.to || [], ...newMsg.recipients.cc || [], ...newMsg.recipients.bcc || []];
       ContactStore.update(undefined, allContacts, { last_use: Date.now() }).catch(Catch.reportErr);
       const body = { 'text/plain': signedData };
-      return await SendableMsg.create(this.acctEmail, { ...this.headers(newMsg), body, atts });
+      //return await SendableMsg.create(this.acctEmail, { ...this.headers(newMsg), body, atts });
+      return await SendableMsg.createPgpInline(this.acctEmail, this.headers(newMsg), body, atts);
     }
     // pgp/mime detached signature - it must be signed later, while being mime-encoded
     // prepare a sign function first, which will be used by Mime.encodePgpMimeSigned later
     const body: SendableMsgBody = { 'text/plain': newMsg.plaintext, 'text/html': newMsg.plainhtml };
-    const sendable = await SendableMsg.create(this.acctEmail, { ...this.headers(newMsg), body, atts, type: 'pgpMimeSigned' });
+    //const sendable = await SendableMsg.create(this.acctEmail, { ...this.headers(newMsg), body, atts, type: 'pgpMimeSigned' });
+    //const sendable = await SendableMsg.createOpenPGPSigned(this.acctEmail, { ...this.headers(newMsg), body, atts });
+    const sendable = await SendableMsg.createPgpMimeSigned(this.acctEmail, this.headers(newMsg), body, atts);
     sendable.setSignMethod((signable: string) => MsgUtil.sign(signingPrv, signable, true));
```

Comment:
These two lines are interesting. `setSignMethod` is only ever used for `createPgpMimeSigned`. Therefore I _think_ you could just pass this method straight into `createPgpMimeSigned` as in `SendableMsg.createPgpMimeSigned(acctEmail, headers, body, atts, (signable: string) => MsgUtil.sign(signingPrv, signable, true))`. This will simplify `SendableMsg` class too, and will overall be simpler.

Reply:
Ok